README.md

```diff
@@ -12,11 +12,113 @@ A plugin for Terraform enabling it to manipulate
+        Type:     schema.TypeString,
+        Optional: true,
+        Description: "Location of the fleet API if --driver=api. Alternatively, if --driver=etcd, location of the etcd API.",
+        Default: "unix:///var/run/fleet.sock",
+    },
+    "etcd-key-prefix": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Default: registry.DefaultKeyPrefix,
+        Description: "Keyspace for fleet data in etcd (development use only!)",
+    },
+    "key-file": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Default: "/var/run/fleet.sock",
+        Description: "Location of TLS key file used to secure communication with the fleet API or etcd",
+    },
+    "cert-file": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Description: "Location of TLS cert file used to secure communication with the fleet API or etcd",
+    },
+    "ca-file": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Description: "Location of TLS CA file used to secure communication with the fleet API or etcd",
+    },
+    "tunnel": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Description: "Establish an SSH tunnel through the provided address for communication with fleet and etcd.",
+    },
+    "known-hosts-file": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Description: "File used to store remote machine fingerprints. Ignored if strict host key checking is disabled.",
+        Default: ssh.DefaultKnownHostsFile,
+    },
+    "ssh-username": &schema.Schema{
+        Type:     schema.TypeString,
+        Optional: true,
+        Description: "Username to use when connecting to CoreOS instance.",
+        Default: "core",
+    },
+
+    "strict-host-key-checking": &schema.Schema{
+        Type:     schema.TypeBool,
+        Optional: true,
+        Description: "Verify host keys presented by remote machines before initiating SSH connections.",
+        Default: true,
+    },
+
+    "ssh-timeout": &schema.Schema{
+        Type:     schema.TypeFloat,
+        Optional: true,
+        Description: "Amount of time in seconds to allow for SSH connection initialization before failing.",
+        Default: 10.0,
+    },
+    "request-timeout": &schema.Schema{
+        Type:     schema.TypeFloat,
+        Optional: true,
+        Description: "Amount of time in seconds to allow a single request before considering it failed.",
+        Default: 3.0,
+    },
+}
+```
+
+EX:
+
+```
+provider "fleet" {
+  driver = "etcd"
+  endpoint = "http://192.168.0.1:4001"
+  // etcd_key_prefix can be ommited to use the default value
+  etcd_key_prefix = "/_some/_weird/etcd/prefix"
+  // connection_retries defaults to 12
+  connection_retries = 9000
+}
+```
+
+```
+provider "fleet" {
+  driver = "api"
+  endpoint = "http://192.168.0.1:8080"
+}
+```
```

Comment:
I think I'd rather not simply copy & paste the provider configuration in the README. Documentation is good, but anyone who wanted to see the code would just read it. I'd bet this copy of the code will quickly become outdated anyhow.

I'd use the Terraform provider documentation as an example for how to format the documentation. Most people reading are probably already familiar with that format.

Also, I suspect connecting via an SSH tunnel is a common case for anyone running Terraform remote to Fleet, so that would be a good example to maintain.